OpenRA.Mods.Cnc/Traits/Minelayer.cs

```diff
@@ -197,13 +210,21 @@ public bool IsCellAcceptable(Actor self, CPos cell)
 			if (!self.World.Map.Contains(cell))
 				return false;
 
+			if (!self.World.ActorMap.GetActorsAt(cell).All(a => a == self))
+				return false;
```

Comment:
We need to be careful with adding this here. a) actors can move while we are running the activity (and this check exists inside the activity right before we lay the mine anyway) and b) this might be leaking information from actors below FoW. I would just drop this change again.

Reply:
Hmm well for the CA use case this was really the main reason for the change, as we have an infantry unit that can lay mines, and you're not able to if you have more than one of them in the same cell (or if there's a mine already there), but the cursor suggests that you can.

When dragging a minefield in the fog it doesn't leak info that way (as there's checks for that which are prioritised in MinefieldOrderGenerator), but it does cause the minelayer to not carry out the order if it "knows" there are actors blocking where you target the minefield, so that does need fixing.

So maybe this should still be checked when deciding which cursor to display, but not for everything else?

Reply:
Can we run this check only when we are targeting the cell we are standing in (deploy cursor)? This cell is certainly visible and we wouldn't leak info.